### ISSUE TITLE
Have a good workflow around precompiled nif usage and rust tools integration

### ISSUE THREAD

**Issue by polvorin** [build-trust/ockam#6091]
The nif depends on the ockam' rust code,  ideally we _want_ it to be compiled during tests (at least if something changed in implementations/rust/* or the nif itself).   But just once,  right now if we don't download it from a predefined location it would be recompiled separately by every elixir "app" (`ockam`,  `ockam_cloud_node`,  `ockam_services`,  etc), making it very slow.

The obvious downside of current approach of downloading the precompiled .so from a well-known location is that while it do speed up test.. it is testing with a fixed, pre-compiled version of the nif at some point on the past, that doesn't correspond with the actual code we have.  And of course we need to manage that artifact as well. 

Some of the above is addressed on 
https://github.com/build-trust/ockam/pull/6093

The rust part of the nif (`native/*`) can be moved to to implementations/rust (see `path` option on https://hexdocs.pm/rustler/Rustler.html) ,  but some investigation is needed to ensure things works as expected. 

Perhaps some ideas could be taken from https://github.com/philss/rustler_precompiled,  altough doubt it could be applied as-is.. due to our usage here (where the rust code the nif depends on,   the nif itself,  and applications using the nif are all part of the same repository). 

To keep in mind precompiled nif is _not_ a critical part for us (it is a nice-to-have one),  but having reasonably duration on our CI test pipeline is (and maybe achiviable by other ways,  as compiling the nif only once, cache the result and somehow reuse it on all targets that runs elixir code).

**Comment by metaclips** [comment]
Closed in https://github.com/build-trust/ockam/pull/7626
